src/lib/components/ui/ScrollView/ScrollView.scss

```diff
@@ -0,0 +1,193 @@
+@import '../../../styles/tools/caret';
+@import '../../../styles/tools/hide-scrollbar';
+@import '../../../styles/tools/reset';
+@import '../../../styles/tools/transitions';
+@import 'theme';
+
+// 1. Scrolling shadows are implemented as pseudo elements. This way we can customise them only
+//    thanks to custom properties.
+// 2. Clip scrolling shadows.
+// 3. Create stacking context just to avoid scrolling shadows being covered by content.
+// 4. Make the content bounding rectangle spread beyond the visible part.
+// 5. Prevent undesired vertical scrolling that may occur with tables inside.
+
+$_arrow-inner-offset: 0.5rem;
+$_arrow-outer-offset: 1rem;
+
+.root {
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  overflow: hidden; // 2.
+
+  // 1.
+  &::before,
+  &::after {
+    @include transition((visibility, opacity, transform));
+
+    content: '';
+    position: absolute;
+    z-index: 1;
+    display: block;
+    visibility: hidden;
+    opacity: 0;
+    width: var(--rui-local-shadow-width);
+    height: var(--rui-local-shadow-height);
+  }
+
+  &::before {
+    background: var(--rui-local-start-shadow-background);
+    box-shadow: var(--rui-local-start-shadow-box-shadow);
+  }
+
+  &::after {
+    background: var(--rui-local-end-shadow-background);
+    box-shadow: var(--rui-local-end-shadow-box-shadow);
+  }
+}
+
+.viewport {
+  position: relative; // 3.
+  width: 100%;
+  scroll-behavior: smooth;
+  -webkit-overflow-scrolling: touch;
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-overflow-scrolling

> Non Standard: This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future.

Is is OK for us?

Reply:
Remove this line, open the demo on your iPhone and you'll see 🕶️ .